core/src/Kokkos_View.hpp

```diff
@@ -774,6 +771,16 @@ class View : public ViewTraits<DataType, Properties...> {
     return m_track.m_tracker;
   }
   //----------------------------------------
+  template <class HandleType>
+  inline void assign_data_handle(HandleType handle) {
+    m_map.m_impl_handle = handle;
+  }
+
+  template <class RecordType>
+  inline void assign_record(RecordType& rec) {
+    m_track.m_tracker.clear();
+    m_track.m_tracker.assign_allocated_record_to_uninitialized(rec);
+  }
```

Comment:
These are new public member functions without impl_ prefix for View: we need to have a really good justification why we should have those. And documentation what they are doing.

Reply:
Commenting to keep it in my head: this was resolved by making the function private. Don't look at this and ask for it to be `impl_` prefixed

Reply:
class now uses attorney to access private members 60b42a7d87599a9d2c2bfe913cb811b38a3b6f9f